### PR TITLE
Change dimension name to following unified IODA naming convention 

### PR DESCRIPTION
**Pull request by emilyhcliu**
## Description
Run IODA-validate to check IODA output for AMSU-A (Normal/DB) and MHS (Normal/DB) and make modifications accordingly

This PR is intended to address "dimension" only:
Change "nchans" to "Channel" and change "nlocs" to "Location"

Working branch /feature/query_validation

Modifications: 
Replace nlocs with Location in src/bufr/IodaEncoder/IodaEncoder.cpp
Replace "nchans" in the YAML with "Channel" in the YAMLs

These modifications caused ctest failure in the followin tests:
```
	468 - test_iodaconv_bufr_mhs2ioda (Failed)
	469 - test_iodaconv_bufr_hrs2ioda (Failed)
	470 - test_iodaconv_bufr_filtering (Failed)
	471 - test_iodaconv_bufr_splitting (Failed)
	472 - test_iodaconv_bufr_filter_split (Failed)
	473 - test_iodaconv_bufr_ncep_1bamua2ioda (Failed)
	474 - test_iodaconv_bufr_ncep_esamua2ioda (Failed)
	475 - test_iodaconv_bufr_ncep_1bmhs2ioda (Failed)
	476 - test_iodaconv_bufr_ncep_esmhs2ioda (Failed)
	477 - test_iodaconv_bufr_adpsfc2ioda (Failed)
	478 - test_iodaconv_bufr_snowadpsfc2ioda (Failed)
	479 - test_iodaconv_bufr_read_2_dim_blocks (Failed)
	480 - test_iodaconv_bufr_read_wmo_radiosonde (Failed)
	481 - test_iodaconv_bufr_satwnd_old_format (Failed)
	482 - test_iodaconv_bufr_satwnd_new_format (Failed)
	483 - test_iodaconv_bufr_sevcrs_v2 (Failed)
	484 - test_iodaconv_bufr_mtiasi (Failed)
```
Test output files are updated.

### CODE REVIEW

**Comment by emilyhcliu** [comment]
Note:
**All test passed except the following**:
	492 - test_iodaconv_satbias_amsua (Failed)
	493 - test_iodaconv_satbias_cris (Failed)
	494 - test_iodaconv_satbias_ssmis (Failed)

The failure is due to the differences in "nchannels" and "npredictors".

Using satbias_amsua_n18 as an example:
The output from the **testrun** has the following:
```
nchannels = 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ;
npredictors = 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ;
```
The output from the **testoutput** (reference) has the following:
```
 nchannels = 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 ;
 npredictors = 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12 ;
```
The rest of the fields are identical between testrun and testoutput.

The feature/query branch should not affect the satbias test result.

File: test/testinput/bufr_filter_split.yaml [hunk 1]
```diff
@@ -50,7 +50,7 @@ observations:
       obsdataout: "./testrun/gdas.t18z.1bmhs.tm00.{splits/hour}.{splits/minute}.filter_split.nc"
 
       dimensions:
-        - name: nchans
+        - name: Chanel
```
Comment (jamiebresch) [inline test/testinput/bufr_filter_split.yaml:53]: typo

Comment (emilyhcliu) [reply test/testinput/bufr_filter_split.yaml]: @jamiebresch Thanks for reviewing this PR.  The type has been corrected and the test output file updated accordingly.

**Comment by PraveenKumar-NOAA** [comment]
All the tests passed except: 

	501 - test_iodaconv_satbias_amsua (Failed)
	502 - test_iodaconv_satbias_cris (Failed)
	503 - test_iodaconv_satbias_ssmis (Failed)

There were errors while validating the IODA output using ioda-validate.x, such as: 

  ERROR: Required group Metadata is using an older name: 'MetaData'.
 Verifying group MetaData
 Verifying group ObsValue
 Verifying group /
 Verifying dimension names
 Verifying variable information
  Variable MetaData/datetime
   Warning: Variable 'MetaData/datetime' uses a superseded name. Replace with 'dateTime'
   ERROR: File variable 'MetaData/datetime' has units of 'datetime', which are invalid.
  Variable MetaData/latitude
  Variable MetaData/longitude
  Variable ObsValue/radiance
   ERROR: Variable 'ObsValue/radiance' has units of 'K', which are not convertible to the YAML-specified units of 'W m-2 sr-1'.

Needed to update the group/variable names, etc.

**Pull request merged**